lib/agent0/agent0/interactive_fuzz/fuzz_present_value.py

```diff
@@ -118,14 +121,37 @@ def fuzz_present_value(
             case HyperdriveActionType.CLOSE_SHORT:
                 maturity_time, open_trade = next(iter(agent.wallet.shorts.items()))
                 trade_event = agent.close_short(maturity_time=maturity_time, bonds=open_trade.balance)
+            case _:
+                raise ValueError(f"Invalid {trade_type=}")
+
+    for trade_type in [HyperdriveActionType.ADD_LIQUIDITY, HyperdriveActionType.REMOVE_LIQUIDITY]:
+        # Keep the agent flush
+        if agent.wallet.balance.amount < FixedPoint("1e10"):
+            agent.add_funds(base=FixedPoint("1e10") - agent.wallet.balance.amount)
+
+        # Set up trade amount bounds
+        min_trade = interactive_hyperdrive.hyperdrive_interface.pool_config.minimum_transaction_amount
+        max_budget = agent.wallet.balance.amount
+        trade_amount = None
+
+        # Execute the trade
+        match trade_type:
             case HyperdriveActionType.ADD_LIQUIDITY:
+                # recompute initial present value for liquidity actions
+                check_data["initial_present_value"] = interactive_hyperdrive.hyperdrive_interface.calc_present_value(
+                    interactive_hyperdrive.hyperdrive_interface.current_pool_state
+                )
                 trade_amount = FixedPoint(
                     scaled_value=int(
                         np.floor(rng.uniform(low=min_trade.scaled_value, high=agent.wallet.balance.amount.scaled_value))
                     )
                 )
                 trade_event = agent.add_liquidity(trade_amount)
             case HyperdriveActionType.REMOVE_LIQUIDITY:
+                # recompute initial present value for liquidity actions
+                check_data["initial_present_value"] = interactive_hyperdrive.hyperdrive_interface.calc_present_value(
```

Comment:
Can you double check this? I think this value is getting overwritten: this loop first executes add liquidity and sets this parameter, then removes liquidity and resets this parameter.

Reply:
the loop recalculates the initial before the trade for each trade; so the steps are
- calculate initial present value
- add liquidity
- check new pv against initial

- calculate initial pv
- remove liquidity
- check new pv against initial

Reply:
I could remove the loops entirely, they are kind of overcomplicating parsing the test

Reply:
Ah okay, I see, didn't realize we check invariances at the end of every loop. In this case, this var should be set outside of the match